### PR TITLE
feat: Cart Functionality

### PR DESCRIPTION
**Pull request by ozairkhan2810**
Added Cart Functionality

- Add to cart
- Remove from cart
- Price details

### CODE REVIEW

File: src/components/cart/cart.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+.wishlist-products {
+  width: 100%;
+}
+
+.card-sub-content {
+  text-align: left;
+}
+
+.cart-quantity-placeholder {
+  margin: 0 0.8rem;
```
Comment (ShreyashMahajan) [inline src/components/cart/cart.css:19]: try to use px for less than 1rem

**Review by ShreyashMahajan — APPROVED** [review]
The code structure looks good.
The code is understandable and clean.
Proper flow is there in the code written

File: src/components/products/ProductList.js [hunk 2]
```diff
@@ -3,12 +3,18 @@ import ProductFilter from "./ProductFilter";
...
 const ProductList = ({ products }) => {
   const {
     wishlistState: { wishlist },
     wishlistDispatch,
   } = useWishlist();
+  const {
+    cartState: { cart },
+    cartDispatch,
+  } = useCart();
+  console.log(cart);
```
Comment (JantuDeb) [inline src/components/products/ProductList.js]: remove console.log

File: src/pages/cart/Cart.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        }
+      : acc;
+  };
+
+  const totalPriceDetails = cart.reduce(priceDetailsReducer, {
+    price: 0,
+    mrp: 0,
+  });
+
+  console.log(totalPriceDetails);
```
Comment (JantuDeb) [inline src/pages/cart/Cart.js]: remove log

File: src/reducer/cart-reducer.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+export const cartReducer = (state, action) => {
+  const { cart } = state;
+  const { type, payload } = action;
+  switch (type) {
+    case "ADD_TO_CART":
+      if (cart.some(({ product }) => product._id === payload._id))
+        return {
+          ...state,
+          cart: [...cart.filter(({ product }) => product._id !== payload._id)],
```
Comment (JantuDeb) [inline src/reducer/cart-reducer.js]: No need to destructure when you apply filter

File: src/reducer/cart-reducer.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+      return {
+        ...state,
+        cart: [...cart, { product: payload, quantity: 1 }],
+      };
+
+    case "REMOVE_FROM_CART":
+      return {
+        ...state,
+        cart: [...cart.filter(({ product }) => product._id !== payload._id)],
```
Comment (JantuDeb) [inline src/reducer/cart-reducer.js]: Same here

**Review by JantuDeb — APPROVED** [review]
Overall it's looks good added few suggestion.

**Review by munjalSaksham71 — APPROVED** [review]
Overall code looks good! Just remove logs and you are good to merge

**Pull request merged**